pkg/config/server/features.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package server
+
+import (
+	"context"
+
+	"github.com/mindersec/minder/internal/auth/jwt"
+)
+
+// FeaturesConfig is the configuration for the features
+type FeaturesConfig struct {
+	// MembershipFeatureMapping maps a membership to a feature
+	MembershipFeatureMapping map[string]string `mapstructure:"membership_feature_mapping"`
+}
+
+// GetFeaturesForMemberships returns the features associated with the memberships in the context
+func (fc *FeaturesConfig) GetFeaturesForMemberships(ctx context.Context) []string {
+	memberships := extractMembershipsFromContext(ctx)
+
+	var features []string
+	for _, m := range memberships {
+		if feature, ok := fc.MembershipFeatureMapping[m]; ok {
```

Comment:
What a about:

```suggestion
		if feature := fc.MembershipFeatureMapping[m]; feature != "" {
```

Which will cover both "not found" and "was an empty string", which we probably don't want.